test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js

```diff
@@ -372,21 +395,61 @@ define([
                     fsmState.NEW
                 );
                 // UI should have been rendered; only one action button visible
-                const actionButtons = this.kbaseNode.querySelectorAll(`.kb-rcp__action-button`);
+                const actionButtons = this.kbaseNode.querySelectorAll(selectors.actionButton);
                 const hiddenButtons = this.kbaseNode.querySelectorAll(
-                    `.kb-rcp__action-button.hidden`
+                    `${selectors.actionButton}.hidden`
                 );
                 expect(actionButtons.length).toEqual(hiddenButtons.length + 1);
             });
+
+            describe('app cell startup states', () => {
+                beforeEach(async function () {
+                    await this.appCellWidgetInstance.init();
+                    await this.appCellWidgetInstance.attach(this.kbaseNode);
+                    await this.appCellWidgetInstance.start();
+                });
+                it('can validate params on startup, params incomplete', async function () {
+                    await this.appCellWidgetInstance.run();
+                    expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).toEqual(
+                        fsmState.EDITING_INCOMPLETE
+                    );
+                    const runButton = this.kbaseNode.querySelector(selectors.run);
+                    expect(runButton).not.toHaveClass('hidden');
+                    expect(runButton).toHaveClass('disabled');
+                });
+
+                it('can validate params on startup and be ready to run', async function () {
+                    // add a valid param so the cell starts up ready to run
+                    this.cell.metadata.kbase.appCell.params.param = 'RAWR!';
+                    await this.appCellWidgetInstance.run();
+                    expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).toEqual(
+                        fsmState.EDITING_COMPLETE
+                    );
+                    const runButton = this.kbaseNode.querySelector(selectors.run);
+                    expect(runButton).not.toHaveClass('hidden');
+                    expect(runButton).not.toHaveClass('disabled');
+                });
+
+                it('can start in an internal error state', async function () {
```

Comment:
I've got some more work to do on starting up in error states for devPoC, so this test will be elaborated upon then.